spec/models/user_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'associations' do
+    it 'has many posts' do
+      association = described_class.reflect_on_association(:posts)
+      expect(association.macro).to eq :has_many
+    end
+
+    it 'has many comments' do
+      association = described_class.reflect_on_association(:comments)
+      expect(association.macro).to eq :has_many
+    end
+
+    it 'has many likes' do
+      association = described_class.reflect_on_association(:likes)
+      expect(association.macro).to eq :has_many
+    end
+  end
+
+  describe 'validations' do
+    it 'validates the presence of name' do
+      user = User.new
+      user.valid?
+      expect(user.errors[:name]).to include("can't be blank")
+    end
+
+    # Uncomment the following block if you want to test bio validation
+    # it 'allows nil for bio' do
+    #   user = User.new(bio: nil)
+    #   user.valid?
+    #   expect(user.errors[:bio]).not_to include("can't be blank")
+    # end
```

Comment:
- Let's kindly remove this commented piece of code to keep our codebase clean and maintainable.